src/clarity/datagrid/datagrid-row.ts

```diff
@@ -36,4 +37,14 @@ export class DatagridRow {
     public set selected(value: boolean) {
         this.selection.setSelected(this.item, value);
     }
+
+    /**
+     * Select / Unselect while click on row
+     */
+    public toggleSelection(element: HTMLElement) {
+        if (this.selection.selectable && element.nodeName !== "INPUT") {
```

Comment:
Working on HTMLElements means you cannot be rendered on the server side, or in webworkers, ... You need to work on Angular nodes and with the Renderer to be completely platform independent.

Admittedly, we still have a few of these in our existing code, but we're removing them as we go. Let's just not add more.